productivity-suite/app/fragments/login/src/components/LoginForm.tsx

```diff
@@ -49,13 +49,16 @@ export const LoginForm = component$(() => {
   });
 
   const dispatchLoginEvent = $(() => {
-    dispatchPiercingEvent(ref.value!, {
-      type: "login",
-      payload: {
+    getBus(ref.value!).dispatch(
+      "login",
+      {
         username: state.username,
         password: state.password,
       },
-    });
+      {
+        singleUse: true,
```

Comment:
I had to add this `singleUse` to make sure that some events get consumed only once

For example the `login` event, if we keep the last login event in the bus, then we wouldn't be able to logout, since on logout the app would go back to the login page, listen for the `login` event, find it in the bus and automatically re-log in.

Reply:
What if we had an `authentication` event, which we use for both login and logout? Then we could send:

```ts
// Login
...dispatch("authentication", {username, password})
```

```ts
// Logout
...dispatch("authentication", null)
```

Reply:
ah, yeah that didn't cross my mind 😅

sounds like a nice solution, I'll apply that change 🙂👍